hyperspy/learn/rpca.py

```diff
@@ -0,0 +1,190 @@
+
+_logger = logging.getLogger(__name__)
+
+def _solveproj(z, X, I, lambda2):
+    m, n = X.shape
+    s = np.zeros(m)
+    x = np.zeros(n)
+    converged = False
+    maxiter = 1e9
+    iter = 0
+
+    ddt = np.dot(scipy.linalg.inv(np.dot(X.T, X) + I), X.T)
+
+    while converged is False:
+        iter += 1
+        xtmp = x
+        x = np.dot(ddt, (z - s))
+        stmp = s
+        s = np.maximum(z - np.dot(X, x) - lambda2, 0.0)
+        stopx = np.sqrt(np.dot(x - xtmp, (x - xtmp).conj()))
+        stops = np.sqrt(np.dot(s - stmp, (s - stmp).conj()))
+        stop = max(stopx, stops) / m
+        if stop < 1e-5 or iter > maxiter:
+            converged = True
+
+    return x, s
+
+def _updatecol(X, A, B, I):
+    tmp, n = X.shape
+    L = X
+    A = A + I
+
+    for i in range(n):
+        b = B[:, i]
+        x = X[:, i]
+        a = A[:, i]
+        temp = (b - np.dot(X, a)) / A[i, i] + x
+        L[:, i] = temp / max(np.sqrt(np.dot(temp, temp.conj())), 1)
+
+    return L
+
+def orpca(X, rank, fast=False, lambda1=None,
+          lambda2=None, method=None,
+          mask=None, seed=None):
+    """
+    This function performs Online Robust PCA with
+    with missing or corrupted data.
+
+    Parameters
+    ----------
+    X : numpy array
+        is the [m x n] matrix of observations.
+    rank : int
+        The model dimensionality.
+    lambda1 : float
+        Nuclear norm regularization parameter.
+    lambda2 : float
+        Sparse error regularization parameter.
+    method : 'CF' | 'BCD'
+        CF  - Closed-form (default)
+        BCD - Block-coordinate descent
+    mask : numpy array
+        is an initial estimate of the sparse error matrix
+    seed : int
+        Random seed for initialization
+
+    Returns
+    -------
+    L : numpy array
+        is the [m x r] basis array
+    R : numpy array
+        is the [r x n] coefficient array
+    E : numpy array
+        is the sparse error
+    U, S, V : numpy arrays
+        are the results of an SVD on Y
+
+    """
+    if fast is True and sklearn_installed is True:
+        def svd(X):
+            return fast_svd(X, rank)
+    else:
+        def svd(X):
+            return scipy.linalg.svd(X, full_matrices=False)
+
+    # Get shape
+    m, n = X.shape
+
+    # Check options if None
+    if method is None:
+        _logger.warning("No method specified. Defaulting to "
+                        "closed-form solver")
+        method = 'CF'
+    if lambda1 is None:
+        _logger.warning("Nuclear norm regularization parameter "
+                        "is set to default.")
+        lambda1 = 1.0 / np.sqrt(n)
+    if lambda2 is None:
+        _logger.warning("Sparse regularization parameter "
+                        "is set to default.")
+        lambda2 = 1.0 / np.sqrt(n)
+
+    # Check options are valid
+    if method not in ('CF', 'BCD'):
+        raise ValueError("'method' not recognised")
+
+    # Set random seed
+    if seed is not None:
```

Comment:
Ah I see, does the seed affect things globally? I don't really understand it in Python.